Data Structures/Heap/BinaryHeap.hpp

```diff
@@ -0,0 +1,334 @@
+#ifndef _HEAP_H_
+#define _HEAP_H_
+
+/*
+*@author Ionésio Junior
+*/
+
+#include <iostream>
+#include <stdlib.h>
+#include "Exception.hpp"
+
+//BinaryHeap.hpp
+
+/* Auxiliar array structure implementation
+*  
+*/
+template <typename T>
+struct Array{
+	int size;
+	T *ptr;
+	bool operator==(Array<T> &other){
+		if(this->size != other.size){
+			return false;
+		}else{
+			bool flag = true;
```

Comment:
## Optimizations could be done
You could write this as
```
			for(int i = 0; i < this->size;i++){
				if(this->ptr[i] != other.ptr[i]){
                                     return false;
				}
			}
                        return true;
```
There is no need for the flag variable at all.